### ISSUE TITLE
Infinite loop when a creature with cycling dies with Abandoned Sarcophagus on the battlefield

### ISSUE THREAD

**Issue by tawmas** [magefree/mage#3619]
It looks like when a creature with cycling dies while Abandoned Sarcophagus is on the battlefield, the client gets stuck in a loop where it repeatedly exiles the creature before the client becomes unresponsive and must be closed forcibly.

I stated the problem above in a very generic way, although in the three instances the creature involved was a Nimble Obstructionist. In the test games I played no other creature with cycling happened to die while I had an Abandoned Sarcophagus on the battlefield.

I believe this only happens when the creature dies to combat damage. I am sure this was the case for the two last instances of the problem, while I did not notice the details of the first crash besides remembering that a Nimble Obstructionist had died since at first I thought about a random problem. While trying to replicate, I noticed that non-creature cards work fine (casting an instant or sorcery with cycling with AS on the battlefield will exile the card after it has finished resolving and the client does not get stuck). I also tried to kill my own Nimble Obstructionist with my own removal while having an AS out, and this also worked correctly.

I took a screenshot of the battlefield, but I did not include the game log because I wanted to attach the saved game log instead, but I discovered too late that the log had not been saved.

**Comment by ingmargoudt** [comment]
Abandoned Sarcophagus was the hardest card to implement of the set, and we just fixed an issue on the cycle ability of Nimble Obstructionist. Will this also occur with any other creature with cycling?

**Comment by tawmas** [comment]
I will try and play a few other test games and try and get another creature killed.

**Comment by tawmas** [comment]
I just managed to have the same problem with an Archfiend of Ifnir.

The message which is repeated in the log is `tawmas moves Archfiend of Ifnir [2af] from the exile zone to the exile zone`. That was the same message I had with the Obstructionist.

During the same game I had another wierd interaction while Abandoned Sarcophagus was out. The AI tapped my Curator of Mysteries with a Stitched Manger, then forced me to sacrifice a creature. At that time I had two creatures on the board: the said Curator and an Archfiend of Ifnir. I chose the Curator. The curator somehow managed to both be exiled (the game log mentioned that it was exiled and that I had sacrificed it, and the Curator appeared in the exile window) and to stay on the board, tapped (and it did not untap on my next untap phase because of the effect of the Stitched Mangler).
